### PR TITLE
handle nil vars for relevant operators

### PR DESCRIPTION
**Pull request by jkim430**
1. sync operator changes (we're currently overriding certain operators in our web repo)
2. handle nil vars for certain operators by returning nil

#2 addresses the scenario where nil vars result in unexpected results - e.g. consider the following rule:
```
{ <: [{ var: 'test' }, 10] }
```
currently, if the 'test' var is nil, this still resolves to true because the operator logic for `<` does `to_f` on all the values, resulting in the comparison `0.0 < 10.0`. we're handling this edge case in `evaluate_conditional` by explicitly setting to false whenever a nil var is involved.

why move this change here now? for calculate step we need to leverage the same `evaluate_conditional` method, but currently it assumes the conditional format of ands within ors. calc step can have arbitrarily nested operators in any order, so we won't be able to parse through it and handle this edge case the same way. moving the logic here should support both cases.

### CODE REVIEW

File: lib/json_logic/operation.rb [hunk 1]
```diff
@@ -128,10 +143,47 @@ def self.is_iterable?(operator)
...
     end
+
+    def self.format_values(values)
+      values = Array(values).flatten
+      # If at least 1 value is numeric, assume all values can be treated as such
+      # Sometimes numbers are represented as strings, so they need to be converted
+      if values.any? { |v| v.is_a?(Numeric) }
+        values.map(&:to_f)
+      elsif values.all? { |v| v.is_a?(String) }
+        values.map(&:downcase)
```
Comment (hmsk) [inline lib/json_logic/operation.rb:182]: might be nice if the cases of `downcase` are tested

**Review by hmsk — APPROVED** [review]

File: lib/json_logic/operation.rb (hunk 1 above)
Comment (jkim430) [reply lib/json_logic/operation.rb:182]: none of the default operators will actually hit this case, but we need this for the custom operators we have in the web repo. those already have specs that test this

Comment (persona-identities) [reply lib/json_logic/operation.rb:182]: I think porting a bare minimum of specs that test `downcase` would be good -- would make it easier to make changes to this repo and not break stuff!

Comment (jkim430) [reply lib/json_logic/operation.rb:182]: ok added some tests for format_values directly https://github.com/persona-id/json-logic-ruby/pull/4
